src/vm/crst.h

```diff
@@ -296,7 +296,11 @@ friend class Crst;
 #endif
 
     union {
+#ifdef CROSSGEN_COMPILE
+        T_CRITICAL_SECTION  m_criticalsection;
```

Comment:
> we will need to bifurcate the CrstBase and some other classes derived from it (CrstStatic, CrstExplicitInit)

I do not think this is needed. crossgen does not have any synchronization because of it is single threaded. Crst operations are implemented as no-opts in crossgen: https://github.com/dotnet/coreclr/blob/master/src/vm/crossgencompile.cpp#L111.

So it is safe to change the Crst data fields to anything we want.

Some comments about it would be useful though.